astropy/io/votable/tree.py

```diff
@@ -2626,6 +2626,12 @@ def _parse_binary(self, mode, iterator, colnumbers, fields, config, pos):
                     mask_bits = careful_read(int((len(fields) + 7) / 8))
                     row_mask_data = list(converters.bitarray_to_bool(
                         mask_bits, len(fields)))
+
+                    # Ignore the mask for string columns (see issue 8995)
+                    for i, f in enumerate(fields):
+                        if f.datatype == 'char' or f.datatype == 'unicodeChar':
```

Comment:
Make the comparison stricter?
```suggestion
                        if row_mask_data[i] and (f.datatype == 'char' or f.datatype == 'unicodeChar'):
```